src/tools/end_quest_tool.py

```diff
@@ -82,20 +82,55 @@ def end_quest(
 
         if not failed:
             # Let's do some things to tidy up.
-            item_name, item_description = generate_quest_item(
-                quest.name, player, context
-            )
 
-            item = Item()
-            item.name = item_name
-            item.description = item_description
+            # matching description (hopefully)
+            quest_description = game_state.quest_arc[len(game_state.quests) - 1]
+
+            new_items = []
+
+            if (
+                quest_description
+                and quest_description.items
+                and len(quest_description.items) > 0
+            ):
+                for prescribed_item in quest_description.items:
+                    item = Item()
+                    item.name = prescribed_item.name
+                    item.description = prescribed_item.description
+                    if item.name:
+                        new_items.append(item)
+
+                    if image_gen := get_item_image_generator(context):
+                        task = image_gen.request_item_image_generation(
+                            item=item, context=context
+                        )
+                        item_image_block = task.wait().blocks[0]
+                        context.chat_history.file.refresh()
```

Comment:
suggestion: I don't think we need this refresh (and it will cost ~100ms or so)

It looks like this was already here, but now is as good a time as any to change it.

Reply:
fwiw, without it, the UI seems to freeze on showing the items.